azurelinuxagent/common/protocol/wire.py

```diff
@@ -133,7 +132,7 @@ def get_vmagent_manifests(self):
     def get_vmagent_pkgs(self, vmagent_manifest):
         goal_state = self.client.get_goal_state()
         ga_manifest = self.client.get_gafamily_manifest(vmagent_manifest, goal_state)
```

Comment:
In the `get_gafamily_manifest` we were first removing the `AGENTS_MANIFEST_FILE_NAME` and then in the `filter_package_list` we were checking if the `AGENTS_MANIFEST_FILE_NAME` exists or not, and if not creating it and returning `ga_manifest.pkg_list`. 
This was unnecessarily just adding a file dependency even when it was not being used. Removed the whole code logic to avoid confusion and always return the `ga_manifest.pkg_list` we receive from GoalSate.

Reply:
i read that code several times and i've no idea what it was trying to do... either there was something very subtle there or it is total ----.  Your approach seems fine. Thanks!